### PR TITLE
ci: implement add/remove issue to/from active sprint feature

### PR DESCRIPTION
**Pull request by yangchiu**
ci: implement add/remove issue to/from active sprint feature

For https://github.com/longhorn/longhorn/issues/4720

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: github-bot/longhorn_github_bot/__init__.py [hunk 1]
```diff
@@ -60,20 +68,156 @@ def issue_transfer(form):
...
-        app.logger.debug('to_pipeline is {}, ignoring'.format(pipeline))
+
+    to_pipeline = form.get('to_pipeline_name')
+    from_pipeline = form.get('from_pipeline_name')
+
+    if to_pipeline == ZENHUB_PIPELINE_ADD_CHECKLIST:
+        add_checklist(issue_number)
+
+    add_to_sprint_pipelines = {ZENHUB_PIPELINE_TRIAGE, ZENHUB_PIPELINE_ICEBOX, ZENHUB_PIPELINE_BACKLOG}
+    remove_from_sprint_pipelines = {ZENHUB_PIPELINE_TRIAGE, ZENHUB_PIPELINE_ICEBOX}
```
Comment (innobead) [inline github-bot/longhorn_github_bot/__init__.py]: Suggest moving these variables as global variables or consts.

File: github-bot/longhorn_github_bot/__init__.py [hunk 2]
```diff
@@ -60,20 +68,156 @@ def issue_transfer(form):
...
+    if from_pipeline in add_to_sprint_pipelines and to_pipeline not in add_to_sprint_pipelines:
+        add_to_sprint(issue_number)
```
Comment (innobead) [inline github-bot/longhorn_github_bot/__init__.py]: Why need to check `from_pipeline`?

File: github-bot/longhorn_github_bot/__init__.py [hunk 3]
```diff
@@ -60,20 +68,156 @@ def issue_transfer(form):
...
+    elif from_pipeline not in remove_from_sprint_pipelines and to_pipeline in remove_from_sprint_pipelines:
+        remove_from_sprint(issue_number)
```
Comment (innobead) [inline github-bot/longhorn_github_bot/__init__.py]: Why need to check `from_pipeline`?

File: github-bot/longhorn_github_bot/__init__.py [hunk 4]
```diff
@@ -16,7 +17,11 @@
 FLASK_USERNAME = generate_password_hash(config('flask_username'))
 GITHUB_OWNER = config('github_owner')
 GITHUB_REPOSITORY = config('github_repository')
-ZENHUB_PIPELINE = config('zenhub_pipeline')
+ZENHUB_PIPELINE_ADD_CHECKLIST = config('zenhub_pipeline')
```
Comment (innobead) [inline github-bot/longhorn_github_bot/__init__.py]: Suggest changing `zenhub_pipeline` to make it understandable.

**Review by innobead — CHANGES_REQUESTED** [review]
In general LGTM. Some comments need to rsolve.

File: github-bot/longhorn_github_bot/__init__.py (hunk 2 above)
Comment (yangchiu) [reply github-bot/longhorn_github_bot/__init__.py]: It's the requirement described in https://github.com/longhorn/longhorn/issues/4720:
```
Add issues to the current sprint if the pipeline changes from Triage, IceBox, and Backlog to another state
```

File: github-bot/longhorn_github_bot/__init__.py (hunk 3 above)
Comment (yangchiu) [reply github-bot/longhorn_github_bot/__init__.py]: If it keeps ifself in `Triage` or `IceBox`, we don't need to do anything.

File: github-bot/longhorn_github_bot/__init__.py (hunk 4 above)
Comment (yangchiu) [reply github-bot/longhorn_github_bot/__init__.py]: Refined.

File: github-bot/longhorn_github_bot/__init__.py (hunk 1 above)
Comment (yangchiu) [reply github-bot/longhorn_github_bot/__init__.py]: Refined.

**Review by innobead — APPROVED** [review]

**Pull request merged**